tests/unit/amo/components/TestConfirmationDialog.js

```diff
@@ -1,118 +1,85 @@
-    const props = {
-      i18n: fakeI18n(),
-      onCancel: sinon.stub(),
-      onConfirm: sinon.stub(),
-      ...otherProps,
-    };
-    return shallowUntilTarget(
-      <ConfirmationDialog {...props} />,
-      ConfirmationDialogBase,
-    );
+  function render(props = {}) {
+    return defaultRender(<ConfirmationDialog {...props} />);
   }
 
   it('renders a dialog', () => {
     const message = 'this action is risky, are you sure?';
-    const root = render({ message });
+    render({ message });
 
-    expect(root.find('.ConfirmationDialog-message')).toHaveLength(1);
-    expect(root.find('.ConfirmationDialog-message')).toHaveText(message);
+    expect(screen.getByText(message)).toHaveClass('ConfirmationDialog-message');
 
-    expect(root.find(Button)).toHaveLength(2);
+    const confirmButton = screen.getByRole('button', { name: 'Confirm' });
+    expect(confirmButton).toHaveClass('ConfirmationDialog-confirm-button');
+    expect(confirmButton).toHaveClass('Button--alert');
 
-    const confirmButton = root.find(Button).at(0);
-    expect(confirmButton).toHaveClassName('ConfirmationDialog-confirm-button');
-    expect(confirmButton).toHaveProp('buttonType', 'alert');
-    expect(confirmButton.children()).toHaveText('Confirm');
-
-    const cancelButton = root.find(Button).at(1);
-    expect(cancelButton).toHaveClassName('ConfirmationDialog-cancel-button');
-    expect(cancelButton).toHaveProp('buttonType', 'cancel');
-    expect(cancelButton.children()).toHaveText('Cancel');
+    const cancelButton = screen.getByRole('button', { name: 'Cancel' });
+    expect(cancelButton).toHaveClass('ConfirmationDialog-cancel-button');
+    expect(cancelButton).toHaveClass('Button--cancel');
   });
 
   it('renders a custom className', () => {
     const className = 'MyComponent';
-    const root = render({ className });
-    expect(root).toHaveClassName(className);
-    expect(root).toHaveClassName('ConfirmationDialog');
+    render({ className });
+    expect(screen.getByClassName('ConfirmationDialog')).toHaveClass(className);
   });
 
-  it.each([true, false])(
-    'can configure buttons as puffy=%s',
-    (puffyButtons) => {
-      const root = render({ puffyButtons });
-
-      expect(root.find('.ConfirmationDialog-cancel-button')).toHaveProp(
-        'puffy',
-        puffyButtons,
-      );
-      expect(root.find('.ConfirmationDialog-confirm-button')).toHaveProp(
-        'puffy',
-        puffyButtons,
-      );
-    },
-  );
-
-  it('calls onConfirm() when user clicks Confirm', () => {
-    const onConfirm = sinon.spy();
-    const root = render({ onConfirm });
+  it('can configure buttons as puffy', () => {
+    render({ puffyButtons: true });
 
-    const event = createFakeEvent();
-    root.find('.ConfirmationDialog-confirm-button').simulate('click', event);
+    const cancelButton = screen.getByRole('button', { name: 'Cancel' });
+    expect(cancelButton).toHaveClass('ConfirmationDialog-cancel-button');
```

Comment:
We have already asserted about this class name in `it('renders a dialog'`, so there's not need to repeat the assertion here. This is true for both buttons in this test and also in the negative puffy buttons test.

Reply:
I think it's worth keeping to ensure that it stays even if we pass `puffyButtons` ? As a guarantee that it doesn't have nasty side effects.